core/controlplane/config/encrypted_assets.go

```diff
@@ -342,6 +379,10 @@ func ReadOrEncryptAssets(dirname string, manageCertificates bool, encryptor Cach
 			{"etcd-client.pem", &r.EtcdClientCert, nil},
 			{"etcd-client-key.pem", &r.EtcdClientKey, nil},
 		}...)
+
+		if caKeyRequiredOnController {
+			files = append(files, entry{"ca-key.pem", &r.CAKey, nil})
```

Comment:
Basically, I've made it try to read `credentials/ca-key.pem` only when it is required.
I'm assuming that it is required when `manageCerticates` and `experimental.tlsBootstrap.enabled` are both set to true, i.e. `ca-key.pem` must be installed on controller nodes for the TLS bootstrapping to work.

@danielfm WDYT?

Reply:
Isn't `ca-key.pem` required for TLS Bootstrapping regardless of using kube-aws-managed certs?

Reply:
@danielfm Yes, `ca-key.pem` is required for bootstrapping even if `manageCertificates`  is set to `true`.
In that case, however, it isn't kube-aws' job to deploy credential files to nodes as it is what `manageCertificates: false` is meant for.

Reply:
```
$ kube-aws version
kube-aws version c22108019ad2ae935e045ccaf4df84282965a0eb
$ kube-aws render credentials --ca-cert-path=./credentials.orig/ca.pem --ca-key-path=./credentials.orig/ca-key.pem
Generating credentials...
-> Parsing existing TLS CA
-> Generating new assets
Error: open credentials/ca-key.pem: no such file or directory
```

Reply:
Ideally, I want `ManageCertificates: true` so the issued certificates are distributed by kube-aws for me. Once they are issued using the ca-key.pem specified as an external asset in a `kube-aws render credentials` command, only the ca.pem should be required for distribution in order for the issued certs to be trusted.

Reply:
@zmt Thx for the feedback!

Although I see your output from `kube-aws version` does show the commit id from this PR, I can say that you might be running an older version of `kube-aws` binary according to the logged messages.
Could you confirm your kube-aws binary is property placed? Probably rebuilding kube-aws and running the kube-aws command by specifying its absolute path would also fix the issue.

Assuming you have:

```yaml
# These are the defaults
manageCertificates: true
experimental:
  tlsBootstrap:
    enabled: false
```

in your `cluster.yaml`, `./credentials.orig/ca-key.pem` should have copied to `credentials/ca-key.pem`.

Reply:
The trick is that I've mistakenly moved `fmt.Println("Generating credentials...")` which was formerly in front of `-> Generating new assets` to to behind it 😅 

So, if you ran the correct binary, you should have seen:

```
-> Generating new assets
Generating credentials...
-> Parsing existing TLS CA
```

whereas what you saw was:

```
Generating credentials...
-> Parsing existing TLS CA
-> Generating new assets
```

Reply:
Oops! I guess I had an old build of the binary lying around. I will be sure the version matches HEAD of this PR fork and confirm it works for me.

Reply:
I rebuilt to find I had `kube-aws version 30b65adf939fc00571450c7cd02a336b3567b2df+dirty`. I had some hacks from months ago that were .gitignored. I removed those:
```
12:24 $ git status -u --ignored
On branch fix-existing-ca-support
Your branch is up-to-date with 'mumoshu/fix-existing-ca-support'.

Ignored files:
  (use "git add -f <file>..." to include in what will be committed)

	core/controlplane/config/files.go
	core/controlplane/config/templates.go
	core/nodepool/config/templates.go
	core/root/config/templates.go

nothing to commit, working tree clean
12:24 $ rm -rf core/controlplane/config/files.go core/controlplane/config/templates.go core/nodepool/config/templates.go core/root/config/templates.go 
12:26 $ git status -u --ignored
On branch fix-existing-ca-support
Your branch is up-to-date with 'mumoshu/fix-existing-ca-support'.

nothing to commit, working tree clean
12:26 $ make
./build
Building kube-aws 30b65adf939fc00571450c7cd02a336b3567b2df
```
Summary of cluster.yaml:
```
12:41 $ grep -Ev '^#|^[         ]*$' cluster.yaml | grep -v '^[         ]*#' | sed -e 's/^\(keyName:\).*/\1 somekey/' -e 's/^\(kmsKeyArn: "arn:aws:kms:us-west-1\).*/\1:012345678901:key\/uuid"/'
clusterName: kms-verif-west-name
apiEndpoints:
- # The unique name of this API endpoint used to identify it inside CloudFormation stacks or
  name: default
  dnsName: kms-verif-west-endpoint
  loadBalancer:
    createRecordSet: false
keyName: somekey
region: us-west-1
availabilityZone: us-west-1c
kmsKeyArn: "arn:aws:kms:us-west-1:012345678901:key/uuid"
worker:
  nodePools:
    - # Name of this node pool. Must be unique among all the node pools in this cluster
      name: nodepool1
addons:
  clusterAutoscaler:
    enabled: false
  rescheduler:
    enabled: false
  tlsBootstrap:
    enabled: false
```

But I still see the error:
```
12:21 $ kube-aws version
kube-aws version 30b65adf939fc00571450c7cd02a336b3567b2df
12:26 $ kube-aws render credentials --ca-cert-path=./credentials.orig/ca.pem --ca-key-path=./credentials.orig/ca-key.pem
Generating credentials...
-> Parsing existing TLS CA
-> Generating new assets
Error: open credentials/ca-key.pem: no such file or directory
12:28 $ ls credentials
admin-key.pem			dex-key.pem			etcd.pem
admin.pem			dex.pem				kubelet-tls-bootstrap-token
apiserver-key.pem		etcd-client-key.pem		tokens.csv
apiserver.pem			etcd-client.pem			worker-key.pem
ca.pem				etcd-key.pem			worker.pem
```

Reply:
@zmt Thanks!

Unfortunately, I'm still unable to reproduce the problem.
For me, kube-aws `30b65adf939fc00571450c7cd02a336b3567b2df` does work in an use-case like yours without any error.

Could you provide me all the sequence of commands from the beginning to the end, i.e. from `kube-aws init` to `kube-aws render credentials`?

Reply:
@zmt I've pushed one more commit to help debugging.

Given a kube-aws binary built from that commit, you will see something like the below from the render credentials command:

```
$ kube-aws render credentials --ca-key-path=./my-ca-key.pem --ca-cert-path=./my-ca.pem
Generating credentials...
-> Parsing existing TLS CA
-> Generating new assets
--> Summarizing the configuration
    Kubelet TLS bootstrapping enabled=false, TLS certificates managed by kube-aws=true, CA key required on controller nodes=false
--> Writing to the storage
--> Verifying the result
```